tests/test_bulk_upload.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+import os
+from instructor import Instructor
+from data import is_valid_student_id, is_valid_grade
+
+def test_valid_student_id():
+    """Test student ID validation"""
+    assert is_valid_student_id("123") == True
+    assert is_valid_student_id("12") == False   # Too short
+    assert is_valid_student_id("abc") == False  # Not digits
+
+def test_valid_grade():
+    """Test grade validation"""
+    assert is_valid_grade("A") == True
+    assert is_valid_grade("B+") == True
+    assert is_valid_grade("X") == False  # Invalid grade
+
+def test_bulk_upload(tmp_path):
+    """Test full CSV upload"""
+    # Create a test CSV file
+    test_file = os.path.join(tmp_path, "test.csv")
+    with open(test_file, 'w') as f:
+        f.write("student_id,grade\n123,A\n456,B+")
```

Comment:
Instead of writing the test_file in the method, add a test file in the folder. It would be easier for others to get an idea of how the bulk upload should work.